src/main/java/com/google/sps/data/Guide.java

```diff
@@ -49,4 +78,25 @@ public String getCategory() {
     public long getTimestamp() {
         return timestamp;
     }
+
+    private boolean nullOrEmptyString(String str){
+        if(str == null) return true;
+        if(str.equals("")) return true;
+        return false;
+    }
+
+    private boolean smallerThan1500bytes(String str){
+        if(title.getBytes().length > 1500) return false;
+        return true;
+    }
+
+    private boolean validIndexedString(String str){
+        if(nullOrEmptyString(str) && smallerThan1500bytes(str)) return false;
```

Comment:
In this case, it would be helpful for the error message to say that it exceeded 1500 bytes, and not just that it was invalid.